unit_test/fuzzing/test_transport/test_spdm_transport_storage_encode_message/spdm_transport_storage_encode_message.c

```diff
@@ -0,0 +1,317 @@
+                                                   message_size,
+                                                   message,
+                                                   &transport_message_size,
+                                                   &transport_message);
+    LIBSPDM_ASSERT(ret == LIBSPDM_STATUS_INVALID_MSG_SIZE);
+    USE_VAR(ret);
+    USE_VAR(hdr);
+}
+
+void libspdm_test_transport_storage_encode_management_cmd(void **state)
+{
+    libspdm_test_context_t *spdm_test_context = *state;
+    storage_spdm_transport_header *hdr;
+    size_t transport_message_size;
+    void *transport_message;
+    size_t allocation_len = 0;
+    uint8_t cmd_direction;
+    uint8_t transport_operation;
+    libspdm_return_t ret;
+
+    if (m_libspdm_transport_storage_test_context.test_buffer_size <
+        sizeof(storage_spdm_transport_header)) {
+        LIBSPDM_ASSERT(false);
+    }
+
+    transport_message_size = LIBSPDM_MAX_SENDER_RECEIVER_BUFFER_SIZE;
+    transport_message = spdm_test_context->test_buffer;
+    cmd_direction = LIBSPDM_STORAGE_CMD_DIRECTION_IF_RECV;
+    transport_operation = SPDM_STORAGE_OPERATION_CODE_DISCOVERY;
+
+    /* Valid Parameters: Test IF_RECV Discovery */
+    ret = libspdm_transport_storage_encode_management_cmd(
+        cmd_direction,
+        transport_operation,
+        0,
+        &transport_message_size,
+        &allocation_len,
+        transport_message);
+    LIBSPDM_ASSERT(ret == LIBSPDM_STATUS_SUCCESS);
+    LIBSPDM_ASSERT(allocation_len != 0);
+
+    /* Trivial Compliance Assertions */
+    hdr = transport_message;
+    LIBSPDM_ASSERT(hdr->security_protocol == SPDM_STORAGE_SECURITY_PROTOCOL_DMTF);
+    LIBSPDM_ASSERT((hdr->security_protocol_specific >> 8) == 0); /* SPSP1 */
+    LIBSPDM_ASSERT((hdr->security_protocol_specific & 0xFF) != 0); /* SPSP0 */
+    LIBSPDM_ASSERT(hdr->inc_512 == false);
+    LIBSPDM_ASSERT(hdr->length == sizeof(storage_spdm_transport_header));
+
+    /* Valid Parameters: Test IF_SEND Pending Info */
+    transport_message_size = LIBSPDM_MAX_SENDER_RECEIVER_BUFFER_SIZE;
+    cmd_direction = LIBSPDM_STORAGE_CMD_DIRECTION_IF_SEND;
+    transport_operation = SPDM_STORAGE_OPERATION_CODE_PENDING_INFO;
+    allocation_len = 0;
+
+    ret = libspdm_transport_storage_encode_management_cmd(
+        cmd_direction,
+        transport_operation,
+        0,
+        &transport_message_size,
+        &allocation_len,
+        transport_message);
+    LIBSPDM_ASSERT(ret == LIBSPDM_STATUS_SUCCESS);
+    LIBSPDM_ASSERT(allocation_len == 0);
+
+    /* Trivial Compliance Assertions */
+    hdr = transport_message;
+    LIBSPDM_ASSERT(hdr->security_protocol == SPDM_STORAGE_SECURITY_PROTOCOL_DMTF);
+    LIBSPDM_ASSERT((hdr->security_protocol_specific >> 8) == 0); /* SPSP1 */
+    LIBSPDM_ASSERT((hdr->security_protocol_specific & 0xFF) != 0); /* SPSP0 */
+    LIBSPDM_ASSERT(hdr->inc_512 == false);
+    LIBSPDM_ASSERT(hdr->length == sizeof(storage_spdm_transport_header));
```

Comment:
if storage_spdm_transport_header is not part of message, why we need sizeof here?